plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -201,13 +201,27 @@ export class EventsProcessor {
         return res
     }
 
-    createEvent(
+    async createEvent(
         preIngestionEvent: PreIngestionEvent,
         person: Person,
         processPerson: boolean
-    ): [RawClickHouseEvent, Promise<void>] {
+    ): Promise<[RawKafkaEvent, Promise<void>]> {
         const { eventUuid: uuid, event, teamId, distinctId, properties, timestamp } = preIngestionEvent
 
+        let team = this.teamManager.getCachedTeam(teamId)
```

Comment:
Good point. Have rewritten to avoid any extra `teamManager` calls. Now we set the `project_id` here: https://github.com/PostHog/posthog/pull/25873/files#diff-51ee2dd21f9a70e91c6e2bf4a0c856040b7f5b7708cbea1c5d674fe1bd92b843R190